homeassistant/components/tplink_omada/binary_sensor.py

```diff
@@ -63,6 +60,18 @@ def get_gateway_port_status_sensors(
                     update_func=lambda p: p.wan_connected,
                 ),
             )
+            yield OmadaGatewayPortBinarySensor(
+                coordinator,
+                gateway,
+                GatewayPortBinarySensorConfig(
+                    port_number=port.port_number,
```

Comment:
Side note: Only the constant Items and lambdas should be part of the entity description. The dynamic info should be passed directly to the entity. The entity description should be defined as a constant at the module level.

Reply:
I switched this over to entity descriptions. Hope I got the right end of the stick.